addons/stock/models/stock_move.py

```diff
@@ -571,6 +571,17 @@ def write(self, vals):
         res = super(StockMove, self).write(vals)
         if receipt_moves_to_reassign:
             receipt_moves_to_reassign._action_assign()
+        # when assign picking for inter-warehouse transfers, set the warehouses as partners on pickings
```

Comment:
I would add the warehouse partner here instead https://github.com/odoo/odoo/blob/69c36510f2443ba0f6bf55d5c451b42c1b352b6d/addons/stock/models/stock_rule.py#L279-L281
It make more sense to set the address at the creation.